Develop

OKuvykina opened the pull request:
unit tests for BooksCollector

OKuvykina committed:
Sprint_4
unit tests for BooksCollector

OKuvykina committed:
Sprint_4
unit tests for BooksCollector

irisqul commented on tests.py:
```diff
@@ -1,24 +1,85 @@
+         ]
+    )
+    def test_add_new_book_not_add_book(self, name):
+        collector = BooksCollector()
```
Отлично: в каждом тесте создается собственный экземпляр BooksCollector, но можно лучше - перенести создание экземпляра в фикстуры в отдельный файл conftest, а не повторять это предусловие в каждом тесте

irisqul commented on tests.py:
```diff
@@ -1,24 +1,85 @@
+
+    def test_get_books_genre_is_empty(self):
+        collector = BooksCollector()
+        assert collector.get_books_genre() == {}
```
Нужно исправить: мы тестируем не атрибуты экземпляров класса, а методы приложения. Необходимо добавить шаги в которых будут использованы методы и проверить результат.

irisqul commented on tests.py:
```diff
@@ -1,24 +1,85 @@
+        collector.set_book_genre('Операция Ы', 'Комедии')
+        collector.add_new_book('Том и Джерри')
+        collector.set_book_genre('Том и Джерри', 'Мультфильмы')
+        assert (collector.get_books_with_specific_genre('Ужасы') and collector.get_books_with_specific_genre('Детективы')) not in collector.get_books_for_children()
```
Тут не совсем понимаю идею - наверное и в шагах нужно было создать книги с жанром ужасы и детективы. Иначе проверка и шаги совсем не связаны

irisqul commented on tests.py:
```diff
@@ -1,24 +1,85 @@
\ No newline at end of file
+        collector.add_new_book('Операция Ы')
+        collector.add_book_in_favorites('Операция Ы')
+        collector.delete_book_from_favorites('Операция Ы ')
```
Нужно исправить: разделить тест на два, добавление книги не протестировано осталось

irisqul reviewed (commented):
Привет, Ольга!

В проекте четвертого спринта получилось попрактиковаться в написании юнит-тестов для небольшого приложения. А еще это первый раз на курсе, когда проект сдан в Git - это делает проект ещё более похожим на настоящее рабочее задание. 

В целом - проект выполнен успешно, но есть и что улучшить. Пожалуйста, посмотри комментарии которые я оставила к коду в Git, в начале каждого комментария указано обязателен ли он к исправлению или это на улучшение. 

После того как проработаешь комментарии - возвращай на проверку. Уверена, что ты отлично справишься)

OKuvykina committed:
Sprint_4
added a fixture
edit test_get_books_for_children_not_have_specific_genre_in_list
added test_add_book_in_favorites_success_add_book

OKuvykina commented:
Всё исправила.
Добавила новый коммит.
